proposals/p3848.md

```diff
@@ -0,0 +1,587 @@
+# Lambdas
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+[Pull request](https://github.com/carbon-language/carbon-lang/pull/3848)
+
+<!-- toc -->
+
+## Table of contents
+
+-   [Abstract](#abstract)
+-   [Syntax Overview](#syntax-overview)
+    -   [Succinctly](#succinctly)
+    -   [Alternative Considered: Terse vs Elaborated](#alternative-considered-terse-vs-elaborated)
+-   [Introducer](#introducer)
+    -   [Alternative Considered: Sigil](#alternative-considered-sigil)
+-   [Positional Parameters](#positional-parameters)
+    -   [Positional Parameter Restrictions](#positional-parameter-restrictions)
+        -   [Alternative Considered: Additional Restriction](#alternative-considered-additional-restriction)
+-   [Function Captures](#function-captures)
+    -   [Capture Modes](#capture-modes)
+        -   [Future Work: Reference Captures](#future-work-reference-captures)
+    -   [Default Capture Mode](#default-capture-mode)
+        -   [Alternative Considered: Middle vs Beginning](#alternative-considered-middle-vs-beginning)
+-   [Function Fields](#function-fields)
+-   [Copy Semantics](#copy-semantics)
+-   [Self and Recursion](#self-and-recursion)
+    -   [Alternative Considered: Recursive Self](#alternative-considered-recursive-self)
+-   [Rationale](#rationale)
+
+<!-- tocstop -->
+
+## Abstract
+
+This document proposes a path forward to add lambdas to Carbon. It further
+proposes augmenting function declarations to create a more continuous syntax
+between the two categories of functions. In short, both lambdas and function
+declarations will be introduced with the `fn` keyword. The presence of a name
+distinguishes a function declaration from a lambda expression, and the rest of
+the syntax applies to both kinds. See [Syntax Overview](#syntax-overview) for
+more.
+
+## Syntax Overview
+
+**Proposal**: A largely continuous syntax between lambdas and function
+declarations.
+
+At a high level, lambdas and function declarations will look like the following.
+
+```
+let lambda1: auto = fn => T.Make();
+
+let lambda2: auto = fn []() -> T { return T.Make(); };
```

Comment:
I understand this and the below are just illustrating where the `[]`s and `()`s go, but I suspect we'll want to not accept empty `[]`s at least... Would it make sense to use some pseudo syntax to indicate that there's something inside? Or to have a more examples?